run/action.yaml

```diff
@@ -0,0 +1,128 @@
+name: "Sandard Action Runner"
+description: "Run a Standard action."
+
+inputs:
+  key_path:
+    description: "The path to place the Nix secret key."
+    required: false
+    default: "${{ runner.temp }}/.nix-key"
+  github_pat:
+    description: "GitHub token for accessing private repos."
+    required: false
+    default: ${{ github.token }}
+  nix_key:
+    description: "Secret key used for Nix package signing."
+    required: false
+  s3_key:
+    description: "Key to upload to S3 Buckets with Nix."
+    required: false
+  s3_id:
+    description: "ID to upload to S3 Buckets with Nix."
+    required: false
+  cache:
+    description: "Nix cache to optionally upload artifacts to"
+    # auto refers to the local nix store, so nothing is uploaded by default
+    default: auto
+    required: false
+  builder:
+    description: "Nix store to build on."
+    default: auto
+    required: true
+  json:
+    description: "JSON describing the action, found in the discovery phase."
+    required: true
+  nix_ssh_key:
+    description: "SSH private key to send builds to a remote Nix builder."
+    default: none
+    required: false
+  ssh_known_hosts:
+    description: "SSH known host keys for Nix builders"
+    default: none
+    required: false
+  extra_nix_config:
+    description: "Configuration to append to the nix.conf."
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set Environment
+      run: |
+        delim=$RANDOM
+
+        cat << $delim >> "$GITHUB_ENV"
+        DISC_PATH=${{ runner.temp }}/discovery
+        DISC_ARC_PATH=${{ runner.temp }}/discovery.tar.zstd
+        SSH_AUTH_SOCK=${{ runner.temp }}/ssh_agent.sock
+        NIX_KEY_PATH=${{ inputs.key_path }}
+        BUILDER=${{ inputs.builder }}
+        CACHE=${{ inputs.cache }}
+        $delim
+
+        echo "JSON<<$delim" >> $GITHUB_ENV
+        echo '${{ inputs.json }}' >> $GITHUB_ENV
+        echo "$delim" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Emplace Secrets
+      run: ${{ github.action_path }}/secrets.sh
+      shell: bash
+      env:
+        NIX_SECRET_KEY: ${{ inputs.nix_key }}
+        AWS_ACCESS_KEY_ID: ${{ inputs.s3_id }}
+        AWS_SECRET_ACCESS_KEY: ${{ inputs.s3_key }}
```

Comment:
To fullfill UX expectations, here we need a if condition (only if at least one secret exist).